src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpReferenceHighlighting.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editor.Implementation.ReferenceHighlighting;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.IntegrationTest.Utilities;
+using Microsoft.VisualStudio.IntegrationTest.Utilities.Input;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Roslyn.VisualStudio.IntegrationTests.CSharp
+{
+    [Collection(nameof(SharedIntegrationHostFixture))]
+    public class CSharpReferenceHighlighting : AbstractEditorTest
+    {
+        protected override string LanguageName => LanguageNames.CSharp;
+
+        public CSharpReferenceHighlighting(VisualStudioInstanceFactory instanceFactory)
+            : base(instanceFactory, nameof(CSharpReferenceHighlighting))
+        {
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Classification)]
+        public void Highlighting()
+        {
+            var markup = @"
+class {|definition:C|}
+{
+    void M<T>({|reference:C|} c) where T : {|reference:C|}
+    {
+        {|reference:C|} c = new {|reference:C|}();
+    }
+}";
+            Test.Utilities.MarkupTestFile.GetSpans(markup, out var text, out IDictionary<string, ImmutableArray<TextSpan>> spans);
+            VisualStudio.Editor.SetText(text);
+            Verify("C", spans);
+
+            // Verify tags disappear
+            VerifyNone("void");
+        }
+
+        private void Verify(string marker, IDictionary<string, ImmutableArray<TextSpan>> spans)
+        {
+            VisualStudio.Editor.PlaceCaret(marker, charsOffset: -1);
+            VisualStudio.Workspace.WaitForAsyncOperations(string.Concat(
+               FeatureAttribute.SolutionCrawler,
+               FeatureAttribute.DiagnosticService,
+               FeatureAttribute.Classification,
+               FeatureAttribute.ReferenceHighlighting));
+
+            AssertEx.SetEqual(spans["reference"], VisualStudio.Editor.GetTagSpans(ReferenceHighlightTag.TagId));
+            AssertEx.SetEqual(spans["definition"], VisualStudio.Editor.GetTagSpans(DefinitionHighlightTag.TagId));
```

Comment:
The existing C# test verifies navigation via the next/prev reference command.  We should preserve that.